plugins/StopAutoDraft/class.stopautodraft.plugin.php

```diff
@@ -8,28 +12,36 @@
 Contact Vanilla Forums Inc. at support [at] vanillaforums [dot] com
 */
 
-class StopAutoDraftPlugin extends Gdn_Plugin {
+class StopAutoDraftPlugin extends Gdn_Plugin
+{
+    /** @var \Vanilla\Contracts\ConfigurationInterface */
+    protected $config;
+
+    /** @var \Vanilla\Models\AddonModel  */
+    protected $addonModel;
+
+    /** @var AddonManager  */
+    protected $addonManager;
+
+    public function __construct(
+        \Vanilla\Contracts\ConfigurationInterface $config,
+        \Vanilla\Models\AddonModel $addonModel,
+        AddonManager $addonManager
+    ) {
+        $this->config = $config;
+        $this->addonModel = $addonModel;
+        $this->addonManager = $addonManager;
+        parent::__construct();
+    }
+
+    public function container_init()
+    {
+        $this->config->saveToConfig("Vanilla.Drafts.Autosave", false);
+        $addon = $this->addonManager->lookupAddon("stopautodraft");
+        $this->addonModel->disable($addon);
```

Comment:
This should definitely not be done in container init. I'd recommend sticking it in structure.

Reply:
Done